File: backend/app/api/v1/endpoints/quotes.py
from fastapi import APIRouter

from app.schemas.quote import QuoteCalculateRequest, QuoteCalculateResponse
from app.services.quote_service import QuoteService

router = APIRouter()


@router.post("/calculate", response_model=QuoteCalculateResponse)
def calculate_quote(payload: QuoteCalculateRequest):
    return QuoteService().calculate(payload)

File: frontend/src/api/modules/quotes.ts
import { http } from "@/api/http";
import type { QuoteCalculateRequest, QuoteCalculateResponse } from "@/types/quote";

export async function calculateQuote(payload: QuoteCalculateRequest) {
  const response = await http.post<QuoteCalculateResponse>("/quotes/calculate", payload);
  return response.data;
}
